stacks-signer/src/signerdb.rs

```diff
@@ -857,13 +857,19 @@ impl SignerDb {
     pub fn get_globally_accepted_block_count_in_tenure(
         &self,
         tenure: &ConsensusHash,
-    ) -> Result<i64, DBError> {
+    ) -> Result<u64, DBError> {
         let query = "SELECT COALESCE((MAX(stacks_height) - MIN(stacks_height) + 1), 0) AS block_count FROM blocks WHERE consensus_hash = ?1 AND state = ?2";
         let args = params![tenure, &BlockState::GloballyAccepted.to_string()];
-        let block_count_opt: Option<i64> = query_row(&self.db, query, args)?;
+        let block_count_opt: Option<u64> = query_row(&self.db, query, args)?;
         match block_count_opt {
-            Some(block_count) => Ok(block_count),
-            None => Ok(0),
+            Some(block_count) => {
+                if block_count > 0 {
+                    Ok(block_count)
+                } else {
+                    Err(DBError::NotFoundError)
+                }
```

Comment:
If the type of `block_count_opt` is `Option<u64>`, then `block_count` cannot possibly be negative, and I think it should be allowed to be `0`.

Reply:
i think that the suggestion of @jcnelson was for enforcing the error when a wrong tenure is queried (a 0 effectively means the tenure does not exist)

Reply:
Oh, right. I see. Got it. Okay, will approve. Thanks!